ldap/servers/slapd/mapping_tree.c

```diff
@@ -1355,17 +1317,31 @@ mapping_tree_entry_add_callback(Slapi_PBlock *pb __attribute__((unused)),
     int i;
     backend *be;
 
-    /* WARNING
-     * for adds we don't need to grab the mapping tree global lock,
-     * because the add operation in the tree is atomic because
-     * only one pointer is updated in the tree.
-     * Should the mapping tree stucture change, this  would have to
-     * be checked again
+    /*
+     * We absolutely do need the lock here because we are *changing things*
+     * and pointer updates are not magically atomic ...
```

Comment:
A very minor point: IMHO this comment is a bit cryptic and should be removed.
         It is a response to the erroneous comment you removed rather than a clear message that stands by itself.
         It will be puzzling when someone will read it later on without the removed message context.
         IMHO: You should just remove that comment:
                There is no need to justify that you lock the mapping tree when modifying its data.

Apart this minor point, LGTM

Reply:
That's a good point about the comment. I'll reword it to be clearer, and as much as I trust your review, I think that @mreynolds389 may want to have a look at this issue too for his ack as well 👍 

Thanks for your time!